db-shared/src/main/java/com/splicemachine/db/iapi/reference/PropertyHelper.java

```diff
@@ -0,0 +1,122 @@
+            "derby.authentication.native.passwordLifetimeMillis";
+    // default lifetime (in milliseconds) of a NATIVE password. FOREVER
+    public static final long MILLISECONDS_IN_DAY = 1000L * 60L * 60L * 24L;
+    public static final long AUTHENTICATION_NATIVE_PASSWORD_LIFETIME_DEFAULT = 0L;
+    // threshhold for raising a warning that a password is about to expire.
+    // raise a warning if the remaining password lifetime is less than this proportion of the max lifetime.
+    public static final String  AUTHENTICATION_PASSWORD_EXPIRATION_THRESHOLD =
+            "derby.authentication.native.passwordLifetimeThreshold";
+    public static final double  AUTHENTICATION_PASSWORD_EXPIRATION_THRESHOLD_DEFAULT = 0.125;
+    /**
+     * Property that specifies the name of the hash algorithm to use with
+     * the configurable hash authentication scheme.
+     */
+    public static final String AUTHENTICATION_BUILTIN_ALGORITHM =
+            "derby.authentication.builtin.algorithm";
+    /**
+     * Default value for db.authentication.builtin.algorithm when creating
+     * a new database.
+     */
+    public static final String AUTHENTICATION_BUILTIN_ALGORITHM_DEFAULT =
+            "SHA-256";
+    /**
+     * Alternative default value for db.authentication.builtin.algorithm if
+     * {@link #AUTHENTICATION_BUILTIN_ALGORITHM_DEFAULT} is not available at
+     * database creation time.
+     */
+    public static final String AUTHENTICATION_BUILTIN_ALGORITHM_FALLBACK =
+            "SHA-1";
+    /**
+     * Property that specifies the number of bytes with random salt to use
+     * when hashing credentials using the configurable hash authentication
+     * scheme.
+     */
+    public static final String AUTHENTICATION_BUILTIN_SALT_LENGTH =
+            "derby.authentication.builtin.saltLength";
+    /**
+     * The default value for db.authentication.builtin.saltLength.
+     */
+    public static final int AUTHENTICATION_BUILTIN_SALT_LENGTH_DEFAULT = 16;
+    /**
+     * Property that specifies the number of times to apply the hash
+     * function in the configurable hash authentication scheme.
+     */
+    public static final String AUTHENTICATION_BUILTIN_ITERATIONS =
+            "derby.authentication.builtin.iterations";
+    /**
+     * Default value for db.authentication.builtin.iterations.
+     */
+    public static final int AUTHENTICATION_BUILTIN_ITERATIONS_DEFAULT = 1000;
+    public static final String NO_ACCESS = "NOACCESS";
+    public static final String READ_ONLY_ACCESS = "READONLYACCESS";
+    public static final String FULL_ACCESS = "FULLACCESS";
+
+    public static Set<String> getAllProperties() {
+        Set<String> propertyList = new TreeSet<>();
+        // get configuration fields from class com.splicemachine.db.iapi.reference.Property
+        for (Field field : com.splicemachine.db.iapi.reference.Property.class.getFields()) {
+            try {
+                if(field.getType().equals(String.class))
```

Comment:
I guess this only retrieves `String` fields and leaves out other fields such as `AUTHENTICATION_BUILTIN_SALT_LENGTH_DEFAULT`, is this intentional?

Reply:
yes, because the NAME of the options are strings. things like AUTHENTICATION_BUILTIN_SALT_LENGTH_DEFAULT are possible values, not names.